// @opuscapita/oc-cm-theme/lib/cjs/theme.js
'use strict';

exports.__esModule = true;

var _colors = require('./colors');

var _constants = require('./constants');

/**
 * Keep this alphabetical order if possible
 */
// Certain variable values are temporarily imported from oc-cm-common-styles
exports.default = {
    colors: _colors.colors,
    column: {
        minWidth: '400px'
    },
    contentBackgroundColor: _colors.colors.white,
    formControls: {
        button: {
            backgroundColor: _colors.colors.grey10,
            primaryBackgroundColor: _colors.colors.orange2,
            textColor: _colors.colors.white,
            minWidth: '70px'
        },
        input: {
            borderColor: _colors.colors.grey7,
            focusBorderColor: _colors.colors.yellow,
            height: '34px'
        }
    },
    halfGutterWidth: '0.5rem',
    header: {
        height: _constants.constants.navBarHeight,
        button: {
            height: '2rem'
        }
    },
    gutterWidth: '1rem',
    inputColumn: {
        flexDirection: 'column',
        labelWidth: '225px',
        labelContainerLineHeight: '1.5rem',
        errorContainer: {
            height: '1.5rem'
        },
        requiredIndicator: {
            fontSize: _constants.constants.fontSize.h4
        }
    },
    inputRow: {
        flexDirection: 'row',
        labelWidth: '225px',
        errorContainer: {
            height: '1.5rem'
        },
        requiredIndicator: {
            fontSize: _constants.constants.fontSize.h4
        }
    },
    screenSizes: {
        medium: '40rem',
        large: '64rem',
        xLarge: '75rem'
    },
    text: {
        title: {
            fontSize: _constants.constants.fontSize.medium,
            fontWeight: _constants.constants.fontWeight.bold
        },
        subTitle: {
            fontSize: _constants.constants.fontSize.medium,
            fontWeight: _constants.constants.fontWeight.normal
        },
        error: {
            fontSize: _constants.constants.fontSize.small,
            color: _colors.colors.red
        }
    }
};
