// @opuscapita/oc-cm-theme/lib/cjs/colors.js
'use strict';

exports.__esModule = true;
var descriptions = {
    grey1: ['Button (disabled)'],
    grey2: ['Alert notification background'],
    grey3: ['Edited grid row', 'Disabled grid text', 'Input border'],
    grey4: ['Select hover background', 'Grid highlighted row background', 'Grid header background'],
    grey5: ['Selected grid row background', 'Selected select item background'],
    grey6: ['App background', 'Grid row border'],
    grey7: ['Grid outer border'],
    grey8: ['Button (hover)'],
    grey9: ['Grid text', 'Button background', 'Checkbox border'],
    grey10: ['Button (pressed)'],

    // Oranges
    orange1: ['Priority button (hover)'],
    orange2: ['Priority button background'],
    orange3: ['Priority button (pressed)'],

    // Other colors e.g. graphs
    yellow: ['Warning color'],
    white: ['Content background color'],
    green: ['Success color'],
    black: [],
    petrol: ['Tooltip background color'],
    red: ['Error color'],
    blue: [],
    violet: []
};

var colors = exports.colors = {
    // Greys
    grey1: '#F0F0F0',
    grey2: '#EAEAEA',
    grey3: '#CCCCCC',

    // Bluish greys
    grey4: '#EFF2F4',
    grey5: '#E6E9EB',
    grey6: '#D3DADE',
    grey7: '#A4AFB6',
    grey8: '#77818c',
    grey9: '#67707C',
    grey10: '#585F68',

    // Oranges
    orange1: '#FF7517',
    orange2: '#EC6608',
    orange3: '#d35c0b',

    // Other colors e.g. graphs
    yellow: '#FECA1D',
    white: '#FFFFFF',
    green: '#3AA57B',
    black: '#000000',
    petrol: '#006070',
    red: '#D82515',
    blue: '#16AED6',
    violet: '#943BA3',
    descriptions: descriptions
};
